### DISCUSSION TITLE
Pore size distribution in compacts

### DISCUSSION THREAD

**Discussion by Mithu1993** [PMEAL/porespy#678] [Q&A]
Hi,

I have data of the position (x,y,z) and radius of spheres in a compact (have attached an example file), generated from the DEM. Basically overlapping spheres, with void spaces (pore). Can I use your software to calculate the pore size and generate the pore structure in the compact? If yes, how can I do it.
[PH101_particle_pos_20_1_1e-06.txt](https://github.com/PMEAL/porespy/files/8412287/PH101_particle_pos_20_1_1e-06.txt)

Best regards,
Mithu

**Comment by jgostick** [comment]
Yes, for sure.  You just need to insert the spheres into an image, and go from there. This is not that hard to do, but there is no function to support it at the moment (though perhaps it should be on our todo list).  Anyway, it'd work like this:

PoreSpy requires voxel images, so all the coordinates would need to be scaled from um to voxels.  This means you need to specify a voxel resolution.  Let's say that your domain is 1 mm cubed.  Let's further limit the size of the final image to 500 voxels cubed (a reasonable size for a normal computer to work with).  This means that your voxel resolution would be 2 um / voxel.  Given your spheres are 93 um in diameter, this means that each sphere will be resolved with about 46 voxels across, which is more than enough.  You can probably get away with 10-20 voxels across.

So, you have created an empty image of 500 cubed (all 0's or Falses), so you just scan through your file and put a single pixel (value of 1 or True) at each point, after scale from meters to um to voxels, so a point at 1e-5 m, 3e-5 m would be 10 um and 30 um, which means voxel 5 and 15. After creating this image of points, you can do a binary dilation using a structuring element with a radius of 26 voxels (i.e 46 um).  

To analyze the void space you'd just invert the image (im = ~im) because porespy studies the True values (i.e. foreground).

Hope that helps.

**Reply by Mithu1993** [reply]
Hi J. Gostick,

Thank you for your reply, can I study the images in 3D? Do you have any example script, where you have done something similar?

Best regards,
Mithu

**Reply by jgostick** [reply]
Yes, 3D works too.  Your file has x, y, z coordinates in meters, so you need to interpret those as voxels as I described above.  I used a 2D example, but it applyes in 3D as well.  I don't have a script handy, but it's really not too hard so I have faith you can figure it out.

**Reply by jgostick** [reply]
I have posted a function for generating an image if the sphere centers and their radii are known, here: #680
